contractsV5/FeeHandler.sol

```diff
@@ -1,9 +1,205 @@
+import "./UtilsV5.sol";
+import "./IBurnableToken.sol";
+import "./IERC20.sol";
+
+contract FeeHandler is IFeeHandler, Utils {
+
+    uint constant ETH_TO_BURN = 10**19;
+    uint constant BITS_PER_PARAM = 64;
+
+    IKyberDAO public kyberDAOContract;
+    IKyberNetworkProxy public kyberNetworkProxyContract;
+    address public kyberNetworkContract;
+    IBurnableToken public knc;
+    uint public burnBlockInterval;
+    uint public lastBurnBlock;
+    uint public brrAndEpochData;
+    // Todo: combine totalRebates and totalRewards into one variable
+    uint public totalRebates;
+    mapping(address => uint) public totalRebatesPerRebateWallet;
+    uint public totalRewards;
+    mapping(uint => uint) public totalRewardsPerEpoch;
+
+    constructor(
+        IKyberDAO _kyberDAOContract,
+        IKyberNetworkProxy _kyberNetworkProxyContract,
+        address _kyberNetworkContract,
+        IBurnableToken _knc,
+        uint _burnBlockInterval
+    ) public
+    {
+        require(address(_kyberDAOContract) != address(0), "The KyberDAO contract cannot be the null address");
+        require(address(_kyberNetworkProxyContract) != address(0), "The KyberNetworkProxy contract cannot be the null address");
+        require(address(_kyberNetworkContract) != address(0), "The KyberNetwork contract cannot be the null address");
+        require(address(_knc) != address(0), "The KNC token contract cannot be the null address");
+
+        kyberDAOContract = _kyberDAOContract;
+        kyberNetworkProxyContract = _kyberNetworkProxyContract;
+        kyberNetworkContract = _kyberNetworkContract;
+        knc = _knc;
+        burnBlockInterval = _burnBlockInterval;
+        lastBurnBlock = block.number;
+        (, uint rewardInBPS, uint rebateInBPS, uint epoch, uint expiryBlock) = kyberDAOContract.getLatestBRRData();
+        brrAndEpochData = encodeData(rewardInBPS, rebateInBPS, epoch, expiryBlock);
+    }
+
+    modifier onlyDAO {
+        require(
+            msg.sender == address(kyberDAOContract),
+            "Only the DAO can call this function."
+        );
+        _;
+    }
+
+    modifier onlyKyberNetwork {
+        require(
+            msg.sender == address(kyberNetworkContract),
+            "Only the internal KyberNetwork contract can call this function."
+        );
+        _;
+    }
+
+    event AccumulateReserveRebate(
+        address[] eligibleWallets,
+        uint[] rebatePercentages,
+        uint totalRebateAmtWei,
+        uint totalRewardAmtWei,
+        uint totalBurnAmtWei
+    );
+
+    // Todo: future optimisation to accumulate rebates for 2 rebate wallet
+    // encode totals, 128 bits per reward / rebate
+    function handleFees(address[] calldata eligibleWallets, uint[] calldata rebatePercentages) external payable onlyKyberNetwork returns(bool) {
+        // Decoding BRR data
+        (uint rewardInBPS, uint rebateInBPS, uint epoch, uint expiryBlock) = decodeData();
+        uint burnInBPS = BPS - rewardInBPS - rebateInBPS;
```

Comment:
burn In bps is actually not used.
can remove it from this function all together.
but in next PR please.

Reply:
done